### PR TITLE
Add e2e DAO + trade test suite running in Docker via GitHub Actions

### PR DESCRIPTION
**Pull request by wodoro**
Introduces a headless Dao gRPC service (passthrough to DaoFacade) covering every proposal type, vote action, blind-vote, reveal, and result inspection, plus a getRawTransaction helper that lets tests deterministically inject Bisq-broadcast txs into bitcoind's mempool. Phase advancement is driven by bitcoin-cli generatetoaddress, so the suite has no wall-clock dependence.

The new apitest package bisq.apitest.dao exercises proposal/blind-vote/ reveal/result phases, full multi-cycle scenarios, vote edge cases (quorum, tie, no-vote, all-ignore, single-voter), and v1 BTC + BSQ-swap trade flows with balance-delta assertions. End-to-end governance loop (propose → vote → result → param flip) is verified via getDaoParamValue.

Stack is 5 containers (bitcoind + seednode + arb + alice + bob) on a docker-compose bridge network. LocalhostNetworkNode now honours BISQ_P2P_LOCALHOST_NAME so each daemon announces a peer-resolvable hostname instead of the hardcoded "localhost".

Workflow .github/workflows/e2e-tests.yml runs the suite on PRs touching core/daemon/cli/proto/apitest/seednode, with Gradle + Docker buildx layer caches to reduce CI time. Pre-existing apitest tests remain @Disabled and are unaffected. (will be addressed in a followup)

## Summary by CodeRabbit

* **New Features**
  * DAO governance: create proposals, set votes, publish blind votes and query cycles/proposals/ballots/vote results/bonded roles/DAO params via CLI and gRPC.

* **Tests**
  * Large end-to-end integration suite covering proposal lifecycles, blind-vote/reveal/result phases, edge cases, dispute-agent bootstrap, and trade scenarios.

* **Documentation**
  * Added guidance for running a regtest Docker stack and baked test data for local E2E testing.

* **Chores / CI**
  * New manual/PR e2e CI workflow and updated ignore rules.

[Review Change Stack](https://app.coderabbit.ai/change-stack/bisq-network/bisq/pull/7802)

### CODE REVIEW

File: apitest/src/test/java/bisq/apitest/dao/DisputeAgentBootstrapTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        // arb daemon is reachable on the host's published port (9997) when running
+        // gradle outside docker; falls back to docker bridge hostname inside container.
+        String host = System.getProperty("apiHost.arb", "localhost");
+        int port = Integer.parseInt(System.getProperty("apiPort.arb", "9997"));
+        String pw = System.getProperty("apiPassword", "xyz");
+        GrpcClient arb = new GrpcClient(host, port, pw);
+
+        // Registration should succeed (or be no-op if already registered).
+        assertDoesNotThrow(() -> arb.registerDisputeAgent("mediator", DEV_KEY));
+        assertDoesNotThrow(() -> arb.registerDisputeAgent("refundagent", DEV_KEY));
```
[the remark this answers is not in the document]
Comment (wodoro) [reply apitest/src/test/java/bisq/apitest/dao/DisputeAgentBootstrapTest.java:49]: not addressing now. create issue for later

File: core/src/main/java/bisq/core/api/CoreDaoService.java [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+        if (cycleIndex < 0 || cycleIndex >= cycles.size()) {
+            throw new IllegalArgumentException(format("invalid cycle index %d (have %d cycles)",
+                    cycleIndex, cycles.size()));
+        }
+        Cycle cycle = cycles.get(cycleIndex);
+        return daoStateService.getEvaluatedProposalList().stream()
+                .filter(ep -> cycleService.isTxInCycle(cycle, ep.getProposal().getTxId()))
+                .map(EvaluatedProposal::getProposal)
+                .collect(java.util.stream.Collectors.toList());
+    }
```
[the remark this answers is not in the document]
Comment (wodoro) [reply core/src/main/java/bisq/core/api/CoreDaoService.java:206]: check latest commit
